T4680 Fixed telegraf prometheus-client listen-address invalid format

ServerForge opened the pull request:
<!-- All PR should follow this template to allow a clean and transparent review -->
<!-- Text placed between these delimiters is considered a comment and is not rendered -->

## Change Summary
<!--- Provide a general summary of your changes in the Title above -->

## Types of changes
<!---
What types of changes does your code introduce? Put an 'x' in all the boxes that apply.
NOTE: Markdown requires no leading or trailing whitespace inside the [ ] for checking
the box, please use [x]
-->
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)
<!-- All submitted PRs must be linked to a Task on Phabricator. -->
https://phabricator.vyos.net/T4680

## Component(s) name
<!-- A rather incomplete list of components: ethernet, wireguard, bgp, mpls, ldp, l2tp, dhcp ... -->
telegraf

## Proposed changes
<!--- Describe your changes in detail -->
There are two current issues with the telegraf prometheus-client listen-address configuration. It seems to accept multiple address values, However it seems to only be able to accept one address in the configuration file, It also does not provide brackets around IPv6 addresses causing a too many colon error when using a single IPv6 Address.

## How to test
<!---
Please describe in detail how you tested your changes. Include details of your testing
environment, and the tests you ran. When pasting configs, logs, shell output, backtraces,
and other large chunks of text, surround this text with triple backtics
```
like this
```
-->
Changes were testested by building vyos-1x package with the changes and then building a vyos image with the modified 1x package. 

Image with changes can be found at https://repo.serverforge.org/vyos/dev/sagitta/vyos-1.4-rolling-202209080518-amd64.iso

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

ServerForge committed:
Fixed prometheus-client listen-address in telegraf.j2 and added include for single listen address.

sever-sever commented on data/templates/telegraf/telegraf.j2:
```diff
@@ -45,7 +45,11 @@
 [[outputs.prometheus_client]]
   ## Address to listen on
+{%     if prometheus_client.listen_address | is_ipv6 %}
+  listen = "[{{ prometheus_client.listen_address if prometheus_client.listen_address is vyos_defined else '' }}]:{{ prometheus_client.port }}"
```
We have “bracketize_ipv6” function
For example 
https://github.com/vyos/vyos-1x/blob/627cfc6d6733998a721ac97f9f5bc44c4c2c6797/data/templates/snmp/etc.snmpd.conf.j2#L48

c-po commented:
Please also adjust your commit headline to include a valid task ID https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md

ServerForge committed:
Merge branch 'vyos:current' into current

ServerForge committed:
telegraf: T4680

Fixed prometheus-client listen-address in telegraf.j2 and added listen-address-single.xml.i for single listen address.

sever-sever commented on data/templates/telegraf/telegraf.j2:
```diff
@@ -45,7 +45,11 @@
 ### Prometheus ###
 [[outputs.prometheus_client]]
   ## Address to listen on
+{%     if prometheus_client.listen_address | is_ipv6 %}
+  listen = "{{ prometheus_client.listen_address | bracketize_ipv6 }}:{{ prometheus_client.port }}"
+{%     else %}
   listen = "{{ prometheus_client.listen_address if prometheus_client.listen_address is vyos_defined else '' }}:{{ prometheus_client.port }}"
+{%     endif %}
```
I guess it should be 
```
listen = "{{ prometheus_client.listen_address | join() | bracketize_ipv6 if prometheus_client.listen_address is vyos_defined else '' }}:{{ prometheus_client.port }}
```

ServerForge replied:
That is much cleaner, il try and update and test it later this evening.

ServerForge committed:
telegraf: T4680
Fixed prometheus-client listen-address in telegraf.j2 and added listen-address-single.xml.i for single listen address.

c-po commented on data/templates/telegraf/telegraf.j2:
```diff
@@ -45,7 +45,7 @@
 ### Prometheus ###
 [[outputs.prometheus_client]]
   ## Address to listen on
-  listen = "{{ prometheus_client.listen_address if prometheus_client.listen_address is vyos_defined else '' }}:{{ prometheus_client.port }}"
```
Why do we need join() if it‘s a single address?

ServerForge replied:
It appears to work fine without the join, it has been removed.

ServerForge committed:
Merge branch 'vyos:current' into current

ServerForge committed:
telegraf: T4680
Fixed prometheus-client listen-address in telegraf.j2 and added listen-address-single.xml.i for single listen address.
